### PR TITLE
Support for building rogue without python

### PR DESCRIPTION
**Pull request by slacrherbst**
This PR provides support for building rogue without python or boost python. This is for systems which will use the rogue data processing c++ libraries but not the variable & device trees in python.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -42,6 +42,21 @@ else()
    message(FATAL_ERROR "Git is required to build rogue!")
 endif()
 
+#####################################
+# Find python3
+#####################################
+set(DO_PYTHON 0)
+
+find_package(PythonInterp 3.6)
```
Comment (ruck314) [inline CMakeLists.txt]: I am assuming that there will be a python 3.7 in the future.  It looks like this check enforce version 3.6 only.  Do we want to be forward compatible with new python versions?

Comment (slacrherbst) [reply CMakeLists.txt]: This is a min version line. But these lines will have to be updated due to Ubuntu packaging styles:

````
# Next try Debian/Ubuntu suffix for boost
if ((NOT Boost_FOUND) AND DO_PYTHON)
   find_package(Boost 1.58 COMPONENTS system thread python-py36)
endif()

# Next try Mac with homebrew boost/python36
if ((NOT Boost_FOUND) AND DO_PYTHON)
   find_package(Boost 1.58 COMPONENTS system thread python36)
endif()
````

File: CMakeLists.txt [hunk 2]
```diff
@@ -19,7 +19,7 @@ if (DEFINED ENV{CONDA_PREFIX})
 endif()
 
 # Check cmake version
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.9)
```
[the remark this answers is not in the document]
Comment (slacrherbst) [reply CMakeLists.txt]: I don't recall but something related to the package searches. What version is in current Ubuntu?

[the remark this answers is not in the document]
Comment (ruck314) [reply CMakeLists.txt]: If I use the Ubuntu's default packages, cmake is 3.5.1.

**Comment by slacrherbst** [comment]
Can you try a rogue build on that with the min value changes to 3.5 and see if it works.

**Pull request merged**